### PR TITLE
fix: `pihole-exporter` NixOS module

### PR DESCRIPTION
**Pull request by drupol**
The `interval` configuration option doesn't exist and has been removed (see [release 0.3.0](https://github.com/eko/pihole-exporter/releases/tag/v0.3.0), and [this PR](https://github.com/eko/pihole-exporter/pull/110) in particular)
The `timeout` configuration option has been added in https://github.com/eko/pihole-exporter/pull/110

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/monitoring/prometheus/exporters/pihole.nix [hunk 1]
```diff
@@ -16,14 +16,6 @@ in
         pi-hole API token which can be used instead of a password
       '';
     };
-    interval = mkOption {
```
Comment (NickCao) [inline nixos/modules/services/monitoring/prometheus/exporters/pihole.nix:19]: Use `lib.mkRemovedOptionModule` to mark the removal.

Comment (NickCao) [reply nixos/modules/services/monitoring/prometheus/exporters/pihole.nix:19]: Also this is release note worthy.

Comment (drupol) [reply nixos/modules/services/monitoring/prometheus/exporters/pihole.nix:19]: Fixed both, thanks for letting me know about that, I've learned something new.

File: nixos/modules/services/monitoring/prometheus/exporters/pihole.nix [hunk 2]
```diff
@@ -6,68 +6,72 @@ let
   cfg = config.services.prometheus.exporters.pihole;
 in
 {
+  imports = [
+    (mkRemovedOptionModule [ "interval"] "This option has been removed.")
+     ({ options.warnings = options.warnings; options.assertions = options.assertions; })
```
Comment (NickCao) [inline nixos/modules/services/monitoring/prometheus/exporters/pihole.nix:11]: What's the purpose of doing so, there are a few other examples of this in nixpkgs, all without explanations.

Comment (drupol) [reply nixos/modules/services/monitoring/prometheus/exporters/pihole.nix:11]: Apparently, this is how to do to add `mkRemovedOptionModule`. There are other examples in the same directory.

Comment (NickCao) [reply nixos/modules/services/monitoring/prometheus/exporters/pihole.nix:11]: The way below is correct, I believe. And there's no need to do that options.warnings thing.
```
mkRemovedOptionModule [ "prometheus" "exporters" "pihole" "interval" ] "This option has been removed."
```

Comment (NickCao) [reply nixos/modules/services/monitoring/prometheus/exporters/pihole.nix:11]: You see that for example: rspamd, doc for the removed option is not correctly rendered on https://search.nixos.org/options?channel=unstable&from=0&size=50&sort=relevance&type=packages&query=services.prometheus.exporters.rspamd

Comment (drupol) [reply nixos/modules/services/monitoring/prometheus/exporters/pihole.nix]: Ok, let's try that now.

Comment (drupol) [reply nixos/modules/services/monitoring/prometheus/exporters/pihole.nix]: It's failing as you can see...

Comment (NickCao) [reply nixos/modules/services/monitoring/prometheus/exporters/pihole.nix]: Asked about this on matrix. Yes you were correct, this is something specific to the prometheus modules... Sorry for the noise.

Comment (NickCao) [reply nixos/modules/services/monitoring/prometheus/exporters/pihole.nix]: This is actually documented in `nixos/modules/services/monitoring/prometheus/exporters.md`, I should have checked that first.

Comment (drupol) [reply nixos/modules/services/monitoring/prometheus/exporters/pihole.nix]: No problem :) I just restored the previous state.

**Comment by NickCao** [comment]
diff LGTM, one last nitpick: the commit message should begin with `nixos/prometheus-pihole-exporter: ...`

**Comment by drupol** [comment]
Done.

**Review by NickCao — APPROVED** [review]

**Comment by drupol** [comment]
Thanks!

**Pull request merged**